pipeline/frontend/yaml/linter/linter.go

```diff
@@ -94,6 +100,23 @@ func (l *Linter) lintFile(config *WorkflowConfig) error {
 	return linterErr
 }
 
+func (l *Linter) lintCloneSteps(config *WorkflowConfig) error {
+	if len(config.Workflow.Clone.ContainerList) == 0 {
+		return nil
+	}
+	var linterErr error
+	for _, container := range config.Workflow.Clone.ContainerList {
+		if !utils.MatchImageExact(container.Image, constant.TrustedCloneImages...) {
+			linterErr = multierr.Append(linterErr,
```

Comment:
Suppose, `woodpeckerci/plugin-git:2.5.3` has been released after WP `3.0` release. Will I be able to use it in my pipeline `clone` step?

Reply:
that's the whole point ... it will just print out an warning if you dont use the allow-listed images, that you wont get the credentials injected for cloning the repo

Reply:
if you dont specify the clone step, this message should never apear or we have a serious bug in the default clone step generation. it just informs users who define an own clone step

Reply:
> you wont get the credentials injected for cloning the repo

So, won't work, right? But it's just new version of official plugin.

> it will just print out an warning

Not related to this PR, but it would be great, if there were two constructors for lint issues: `newLinterWarning` and `newLinterError`.

Reply:
I think this will be crazy to maintain and real pain for users who will want to use latest version (because renovate will update it for example) but will have to ask admin all the time to add it to the list (that would require server restart that might be not easy in working time)

Reply:
the `latest` tag and `next` tag are also valid and allowd by default

> So, won't work, right? But it's just new version of official plugin.
I dont get it ... it woll work if you use the ''official plugin" in any case ...

Reply:
I've just tested this:
```go

{
	from: "{steps: { build: { image: golang, settings: { test: 'true' } } }, when: { branch: main, event: push }, clone: { git: { image: woodpeckerci/plugin-git:2.5.3 } } }",
	want: "Specified clone image does not match allow list, netrc will not be injected",
},
```
and it passed.

Suppose, I use woodpecker version `3.0`. New versions of clone plugin released: `2.5.3` and `2.6.0`. I want to use `2.5.3` in my pipeline.
`2.5.2` (in WP `3.0`) - > `2.5.3` (will work?) -> `2.6.0` (aka `latest`)
In other words, would clone plugin work, if I use versions between the last hard-coded in WP (`2.5.2`) and the last real version (`latest=2.6.0`)?

Reply:
yes 

thats the list: https://github.com/woodpecker-ci/woodpecker/pull/4056/files#diff-0f5e5e5da97281ce5b634780f49f2b7783809e6780f517311aa5a7d1b420af3dR50